Fix: Harden FreeMarker against SSTI

AT190510-Cuong opened the pull request:
### Summary
This pull request hardens the FreeMarker template rendering logic to prevent potential Server-Side Template Injection (SSTI) vulnerabilities.

### Details
- Added strict configuration for FreeMarker to disallow arbitrary expression evaluation.
- Ensured only whitelisted template variables are accessible.
- Added validation for user-controlled inputs before passing to template engine.

### Impact
Prevents attackers from injecting malicious template expressions that could lead to arbitrary code execution.

### References
- OWASP: [Server-Side Template Injection](https://owasp.org/www-community/attacks/Server-Side_Template_Injection)

angelozerr commented:
Ci build fails with your test https://github.com/opensagres/xdocreport/actions/runs/18632685970/job/53438957761#step:4:1797

AT190510-Cuong commented:
Hi @angelorerr,
Sorry for the confusion earlier — I’ve now updated the original PR (#705) with the latest code from fix/ssti_new.
I’ve fixed the issue and really hope it works properly this time 😊
Thank you for your patience and for reviewing again!

angelozerr commented:
@AT190510-Cuong I understand now what you are trying to do, but I don t think it is relevant for XDocReport project.

Indeed I dont see any usecase where a user will write a mergefield with freemarker operator. And even if a report contains this syntax it is the resposability of the application which uses XDocReport which must take care of that.

It is the same thing when you are writting a webapp application which uses freemarker to render html page.

I am sorry to not accept your PR. It is not relevant for XDocReport project.

AT190510-Cuong commented on template/fr.opensagres.xdocreport.template.freemarker/src/main/java/fr/opensagres/xdocreport/template/freemarker/FreemarkerTemplateEngine.java:
```diff
@@ -197,6 +197,16 @@ public void setFreemarkerConfiguration( Configuration freemarkerConfiguration )
+        // Security fix: Block dangerous class instantiation via ?new operator to prevent SSTI attacks
+        try
+        {
+            this.freemarkerConfiguration.setSetting( Configuration.NEW_BUILTIN_CLASS_RESOLVER_KEY, "safer" );
```
@angelozerr Thanks for the question!
`Configuration.NEW_BUILTIN_CLASS_RESOLVER_KEY` was introduced in FreeMarker 2.3.30, so it is supported in the FreeMarker version currently used in XDocReport (2.3.32).
